test/extensions/tracers/zipkin/span_buffer_test.cc

```diff
@@ -265,102 +303,150 @@ TEST(ZipkinSpanBufferTest, SerializeSpan) {
+                                            R"("id":"AQAAAAAAAAA=",)"
+                                            R"("kind":"CLIENT",)"
+                                            R"("timestamp":"DEFAULT_TEST_TIMESTAMP",)"
+                                            R"("duration":"DEFAULT_TEST_DURATION",)"
+                                            R"("localEndpoint":{)"
+                                            R"("serviceName":"service1",)"
+                                            R"("ipv4":"AQIDBA==",)"
+                                            R"("port":8080},)"
+                                            R"("tags":{)"
+                                            R"("component":"proxy"}},)"
+                                            R"({)"
+                                            R"("traceId":"AAAAAAAAAAE=",)"
+                                            R"("id":"AQAAAAAAAAA=",)"
+                                            R"("kind":"SERVER",)"
+                                            R"("timestamp":"DEFAULT_TEST_TIMESTAMP",)"
+                                            R"("duration":"DEFAULT_TEST_DURATION",)"
+                                            R"("localEndpoint":{)"
+                                            R"("serviceName":"service1",)"
+                                            R"("ipv4":"AQIDBA==",)"
+                                            R"("port":8080},)"
+                                            R"("tags":{)"
+                                            R"("component":"proxy"})"
+                                            "}]}"),
             serializedMessageToJson<zipkin::proto3::ListOfSpans>(buffer6.serialize()));
 }
 
+TEST(ZipkinSpanBufferTest, TestSerializeTimestampInTheFuture) {
+  ProtobufWkt::Struct objectWithScientificNotation;
+  auto* objectWithScientificNotationFields = objectWithScientificNotation.mutable_fields();
+  (*objectWithScientificNotationFields)["timestamp"] = ValueUtil::numberValue(
+      DEFAULT_TEST_TIMESTAMP); // the value of DEFAULT_TEST_TIMESTAMP is 1584324295476870.
+  const auto objectWithScientificNotationJson =
+      MessageUtil::getJsonStringFromMessage(objectWithScientificNotation, false, true);
+  // Since we use ValueUtil::numberValue to set the timestamp, we expect to
+  // see the value is rendered with scientific notation (1.58432429547687e+15).
+  EXPECT_EQ(R"({"timestamp":1.58432429547687e+15})", objectWithScientificNotationJson);
+
+  ProtobufWkt::Struct object;
+  auto* objectFields = object.mutable_fields();
+  Util::Replacements replacements;
+  (*objectFields)["timestamp"] = Util::uint64Value(DEFAULT_TEST_TIMESTAMP, replacements);
+  const auto objectJson = MessageUtil::getJsonStringFromMessage(object, false, true);
+  // We still have "1584324295476870" from MessageUtil::getJsonStringFromMessage here.
+  EXPECT_EQ(R"({"timestamp":"1584324295476870"})", objectJson);
```

Comment:
Before replacement, `Util::uint64Value` stores the value as a string. `"1584324295476870"`.